pkg/secrets/cloud/manager.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloud
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	gosecrets "gocloud.dev/secrets"
+	_ "gocloud.dev/secrets/awskms"        // support for awskms://
+	_ "gocloud.dev/secrets/azurekeyvault" // support for azurekeyvault://
+	_ "gocloud.dev/secrets/gcpkms"        // support for gcpkms://
+	_ "gocloud.dev/secrets/hashivault"    // support for hashivault://
+
+	"github.com/pulumi/pulumi/pkg/resource/config"
+	"github.com/pulumi/pulumi/pkg/secrets"
+)
+
+// Type is the type of secrets managed by this secrets provider
+const Type = "cloud"
```

Comment:
What's the idea here?

Reply:
Not sure I follow the question?  This is aligned with the general design of these `secrets/*/manager.go` implementations.

Reply:
Ah, I see how this is used now. Disregard this question.